lib/services/plugins-service.ts

```diff
@@ -0,0 +1,190 @@
+///<reference path="../.d.ts"/>
+"use strict";
+import path = require("path");
+import shelljs = require("shelljs");
+import semver = require("semver");
+import Future = require("fibers/future");
+import constants = require("./../constants");
+let xmlmerge = require("xmlmerge-js");
+
+export class PluginsService implements IPluginsService {
+	private static INSTALL_COMMAND_NAME = "install";
+	private static UNINSTALL_COMMAND_NAME = "uninstall";
+	
+	constructor(private $npm: INodePackageManager,
+		private $fs: IFileSystem,
+		private $projectData: IProjectData,
+		private $platformsData: IPlatformsData,
+		private $projectDataService: IProjectDataService,
+		private $childProcess: IChildProcess,
+		private $options: IOptions,
+		private $logger: ILogger,
+		private $errors: IErrors) { }
+		
+	public add(plugin: string): IFuture<void> {
+		return (() => {
+			let pluginName = this.executeNpmCommand(PluginsService.INSTALL_COMMAND_NAME, plugin).wait();
+			let nodeModuleData = this.getNodeModuleData(pluginName);
+			if(!nodeModuleData.isPlugin) {
```

Comment:
This is indeed an interesting topic. Let's brainstorm of what is a plugin.

I think this is an npm package that has a `nativescript` complex property in its package.json. @ligaz, @teobugslayer, your input?